bindinfo/handle.go

```diff
@@ -444,3 +451,51 @@ func (h *BindHandle) logicalDeleteBindInfoSQL(normdOrigSQL, db string, updateTs
 		normdOrigSQL,
 		db)
 }
+
+// GenHintsFromSQL is used to generate hints from SQL.
+// It is used to avoid the circle dependence with planner package.
+var GenHintsFromSQL func(ctx context.Context, sctx sessionctx.Context, node ast.Node, is infoschema.InfoSchema) (string, error)
+
+// CaptureBaselines is used to automatically capture plan baselines.
+func (h *BindHandle) CaptureBaselines(is infoschema.InfoSchema) {
+	schemas, sqls := stmtsummary.StmtSummaryByDigestMap.GetMoreThanOnceSelect()
+	for i := range sqls {
+		stmt, err := h.parser4Baseline.ParseOneStmt(sqls[i], "", "")
+		if err != nil {
+			logutil.BgLogger().Debug("parse SQL failed", zap.String("SQL", sqls[i]), zap.Error(err))
+			continue
+		}
+		normalizedSQL, digiest := parser.NormalizeDigest(sqls[i])
+		if r := h.GetBindRecord(digiest, normalizedSQL, schemas[i]); r != nil && r.Status == Using {
+			continue
+		}
+		h.sctx.Lock()
+		err = h.sctx.RefreshTxnCtx(context.TODO())
```

Comment:
Why do we need this function call?

Reply:
It will create a new txn and the txn maybe used by the optimizer when generating hints, since it may execute some sub queries, and without it, the test will panic.